avocado/utils/software_manager.py

```diff
@@ -1229,77 +1220,3 @@ def install_distro_packages(distro_pkg_map, interactive=False):
                   detected_distro.name, detected_distro.arch,
                   detected_distro.version, detected_distro.release)
     return result
-
-
-def main():
```

Comment:
I can't comment here. Not sure if this is not used by other projects/users. But if not, I'm in favor of removing this.

Reply:
The `__main__` entry point existed mostly to leverage testing of the functionality when it was developed back in the day. If anything, it should be transferred to an explicit entry point, maybe `avocado-software-manager`, but since it's not being used, removing it is the correct call.

Reply:
Hi @lmr ! yeah, I forgot to mention on PR what I also think it could be converted into an Avocado sub-command, if users need this.

Reply:
Something as simple as:

```diff
diff --git a/setup.py b/setup.py
index f5b71601..5873d387 100755
--- a/setup.py
+++ b/setup.py
@@ -66,6 +66,7 @@ if __name__ == '__main__':
                   'avocado-runner-exec-test = avocado.core.nrunner:main',
                   'avocado-runner-python-unittest = avocado.core.nrunner:main',
                   'avocado-runner-avocado-instrumented = avocado.core.nrunner_avocado_instrumented:main',
+                  'avocado-software-manager = avocado.utils.software_manager:main'
                   ],
               'avocado.plugins.cli': [
                   'envkeep = avocado.plugins.envkeep:EnvKeep',

```

Should do the trick.  Then we can remove only the `if __main__` block, and make this available to more.  Hopefully, it'd be useful.